bbq/compiler/compiler.go

```diff
@@ -3298,21 +3360,53 @@ func (c *Compiler[_, _]) addGlobalsFromImportedProgram(location common.Location)
 	// Add a global variable for the imported contract value.
 	contracts := importedProgram.Contracts
 	for _, contract := range contracts {
-		c.addImportedGlobal(location, contract.Name)
+		name := contract.Name
+		alias, ok := aliases[name]
+		if ok {
+			// we want a table pointing from the alias -> address qualifier (Foo1 -> A.0001.Foo)
+			addressQualifiedName := string(location.TypeID(nil, name))
+			c.globalAliasTable[alias] = addressQualifiedName
+
+			c.globalRemoveAddressTable[addressQualifiedName] = name
+			name = addressQualifiedName
+		}
```

Comment:
This is repeated below for imported variables and functions. Consider refactoring this into a reusable function